app/scripts/metamask-controller.js

```diff
@@ -2189,12 +2190,72 @@ export default class MetamaskController extends EventEmitter {
   }
 
   async getTokenStandardAndDetails(address, userAddress, tokenId) {
-    const details =
-      await this.assetsContractController.getTokenStandardAndDetails(
+    const { tokenList } = this.tokenListController.state;
+    const { tokens } = this.tokensController.state;
+
+    const staticTokenListDetails =
+      STATIC_MAINNET_TOKEN_LIST[address.toLowerCase()] || {};
+    const tokenListDetails = tokenList[address.toLowerCase()] || {};
+    const userDefinedTokenDetails =
+      tokens.find(({ address: _address }) =>
```

Comment:
is `tokens` guaranteed to be defined and an array?

Reply:
good question, I think the answer is yes https://github.com/MetaMask/core/blob/main/packages/assets-controllers/src/TokensController.ts#L107-L108